### PR TITLE
image: use golang-fedora as builder

### LINKED ISSUE #1093: install: ERROR: `version LIBVIRT_QEMU_8.2.0 not found (required by cloud-api-adaptor)`
When deploying cloud-api-adaptor on AKS, I get the following error using latest staging:

```shell
cloud-api-adaptor: /lib64/libvirt-qemu.so.0: version `LIBVIRT_QEMU_8.2.0' not found (required by cloud-api-adaptor)  
```
Did any requirements changed due to the switch in how we deploy CAA?

### LINKED ISSUE #1113: libvirt: cloud-api-adaptor-daemonset pod failing due  `LIBVIRT_QEMU_8.2.0` not found
Running the latest and greatest CAA for libvirt, I got the following error on the cloud-api-adaptor-daemonset pod:

```
$ kubectl logs pods/cloud-api-adaptor-daemonset-49xjw -n confidential-containers-system
+ exec cloud-api-adaptor libvirt -uri 'qemu+ssh://wmoschet@192.168.122.1/system?no_verify=1' -data-dir /opt/data-dir -pods-dir /run/peerpod/pods -network-name default -pool-name default -socket /run/peerpod/hypervisor.sock
cloud-api-adaptor: /lib64/libvirt-qemu.so.0: version `LIBVIRT_QEMU_8.2.0' not found (required by cloud-api-adaptor)
```

It seems that cloud-api-adaptor binary is linked against libvirt-qemu library version 8.2.0 but the container environment doesn't provide that version.

Drilling down, it turns out that the container environment provides libvirt-qemu 8.1.0 instead:
```
[wmoschet@wainer-laptop cloud-api-adaptor]$ docker run --rm -it --entrypoint /bin/bash quay.io/confidential-containers/cloud-a
pi-adaptor
[root@a61c75da8836 /]# rpm -qa | grep libvirt
libvirt-libs-8.1.0-2.fc36.x86_64
[root@a61c75da8836 /]# rpm -q --provides libvirt-libs-8.1.0-2.fc36.x86_64
config(libvirt-libs) = 8.1.0-2.fc36
libvirt-admin.so.0()(64bit)
libvirt-admin.so.0(LIBVIRT_ADMIN_2.0.0)(64bit)
libvirt-admin.so.0(LIBVIRT_ADMIN_3.0.0)(64bit)
libvirt-admin.so.0(LIBVIRT_ADMIN_PRIVATE_8.1.0)(64bit)
libvirt-libs = 8.1.0-2.fc36
libvirt-libs(x86-64) = 8.1.0-2.fc36
libvirt-lxc.so.0()(64bit)
libvirt-lxc.so.0(LIBVIRT_LXC_1.0.2)(64bit)
libvirt-lxc.so.0(LIBVIRT_LXC_1.0.4)(64bit)
libvirt-lxc.so.0(LIBVIRT_LXC_2.0.0)(64bit)
libvirt-qemu.so.0()(64bit)
libvirt-qemu.so.0(LIBVIRT_QEMU_0.10.0)(64bit)
libvirt-qemu.so.0(LIBVIRT_QEMU_0.8.3)(64bit)
libvirt-qemu.so.0(LIBVIRT_QEMU_0.9.4)(64bit)
libvirt-qemu.so.0(LIBVIRT_QEMU_1.2.3)(64bit)
libvirt.so.0()(64bit)
libvirt.so.0(LIBVIRT_0.0.3)(64bit)
libvirt.so.0(LIBVIRT_0.0.5)(64bit)
libvirt.so.0(LIBVIRT_0.1.0)(64bit)
libvirt.so.0(LIBVIRT_0.1.1)(64bit)
libvirt.so.0(LIBVIRT_0.1.4)(64bit)
libvirt.so.0(LIBVIRT_0.1.5)(64bit)
libvirt.so.0(LIBVIRT_0.1.9)(64bit)
libvirt.so.0(LIBVIRT_0.10.0)(64bit)
libvirt.so.0(LIBVIRT_0.10.2)(64bit)
libvirt.so.0(LIBVIRT_0.2.0)(64bit)
libvirt.so.0(LIBVIRT_0.2.1)(64bit)
libvirt.so.0(LIBVIRT_0.2.3)(64bit)
libvirt.so.0(LIBVIRT_0.3.0)(64bit)
libvirt.so.0(LIBVIRT_0.3.2)(64bit)
libvirt.so.0(LIBVIRT_0.3.3)(64bit)
libvirt.so.0(LIBVIRT_0.4.0)(64bit)
libvirt.so.0(LIBVIRT_0.4.1)(64bit)
libvirt.so.0(LIBVIRT_0.4.2)(64bit)
libvirt.so.0(LIBVIRT_0.4.5)(64bit)
libvirt.so.0(LIBVIRT_0.5.0)(64bit)
libvirt.so.0(LIBVIRT_0.6.0)(64bit)
libvirt.so.0(LIBVIRT_0.6.1)(64bit)
libvirt.so.0(LIBVIRT_0.6.3)(64bit)
libvirt.so.0(LIBVIRT_0.6.4)(64bit)
libvirt.so.0(LIBVIRT_0.7.0)(64bit)
libvirt.so.0(LIBVIRT_0.7.1)(64bit)
libvirt.so.0(LIBVIRT_0.7.2)(64bit)
libvirt.so.0(LIBVIRT_0.7.3)(64bit)
libvirt.so.0(LIBVIRT_0.7.5)(64bit)
libvirt.so.0(LIBVIRT_0.7.7)(64bit)
libvirt.so.0(LIBVIRT_0.8.0)(64bit)
libvirt.so.0(LIBVIRT_0.8.1)(64bit)
libvirt.so.0(LIBVIRT_0.8.2)(64bit)
libvirt.so.0(LIBVIRT_0.8.5)(64bit)
libvirt.so.0(LIBVIRT_0.8.6)(64bit)
libvirt.so.0(LIBVIRT_0.8.8)(64bit)
libvirt.so.0(LIBVIRT_0.9.0)(64bit)
libvirt.so.0(LIBVIRT_0.9.10)(64bit)
libvirt.so.0(LIBVIRT_0.9.11)(64bit)
libvirt.so.0(LIBVIRT_0.9.13)(64bit)
libvirt.so.0(LIBVIRT_0.9.2)(64bit)
libvirt.so.0(LIBVIRT_0.9.3)(64bit)
libvirt.so.0(LIBVIRT_0.9.4)(64bit)
libvirt.so.0(LIBVIRT_0.9.5)(64bit)
libvirt.so.0(LIBVIRT_0.9.7)(64bit)
libvirt.so.0(LIBVIRT_0.9.8)(64bit)
libvirt.so.0(LIBVIRT_0.9.9)(64bit)
libvirt.so.0(LIBVIRT_1.0.0)(64bit)
libvirt.so.0(LIBVIRT_1.0.1)(64bit)
libvirt.so.0(LIBVIRT_1.0.2)(64bit)
libvirt.so.0(LIBVIRT_1.0.3)(64bit)
libvirt.so.0(LIBVIRT_1.0.5)(64bit)
libvirt.so.0(LIBVIRT_1.0.6)(64bit)
libvirt.so.0(LIBVIRT_1.1.0)(64bit)
libvirt.so.0(LIBVIRT_1.1.1)(64bit)
libvirt.so.0(LIBVIRT_1.1.3)(64bit)
libvirt.so.0(LIBVIRT_1.2.1)(64bit)
libvirt.so.0(LIBVIRT_1.2.11)(64bit)
libvirt.so.0(LIBVIRT_1.2.12)(64bit)
libvirt.so.0(LIBVIRT_1.2.14)(64bit)
libvirt.so.0(LIBVIRT_1.2.15)(64bit)
libvirt.so.0(LIBVIRT_1.2.16)(64bit)
libvirt.so.0(LIBVIRT_1.2.17)(64bit)
libvirt.so.0(LIBVIRT_1.2.19)(64bit)
libvirt.so.0(LIBVIRT_1.2.3)(64bit)
libvirt.so.0(LIBVIRT_1.2.5)(64bit)
libvirt.so.0(LIBVIRT_1.2.6)(64bit)
libvirt.so.0(LIBVIRT_1.2.7)(64bit)
libvirt.so.0(LIBVIRT_1.2.8)(64bit)
libvirt.so.0(LIBVIRT_1.2.9)(64bit)
libvirt.so.0(LIBVIRT_1.3.3)(64bit)
libvirt.so.0(LIBVIRT_2.0.0)(64bit)
libvirt.so.0(LIBVIRT_2.2.0)(64bit)
libvirt.so.0(LIBVIRT_3.0.0)(64bit)
libvirt.so.0(LIBVIRT_3.1.0)(64bit)
libvirt.so.0(LIBVIRT_3.4.0)(64bit)
libvirt.so.0(LIBVIRT_3.7.0)(64bit)
libvirt.so.0(LIBVIRT_3.9.0)(64bit)
libvirt.so.0(LIBVIRT_4.1.0)(64bit)
libvirt.so.0(LIBVIRT_4.10.0)(64bit)
libvirt.so.0(LIBVIRT_4.4.0)(64bit)
libvirt.so.0(LIBVIRT_4.5.0)(64bit)
libvirt.so.0(LIBVIRT_5.10.0)(64bit)
libvirt.so.0(LIBVIRT_5.2.0)(64bit)
libvirt.so.0(LIBVIRT_5.5.0)(64bit)
libvirt.so.0(LIBVIRT_5.6.0)(64bit)
libvirt.so.0(LIBVIRT_5.7.0)(64bit)
libvirt.so.0(LIBVIRT_5.8.0)(64bit)
libvirt.so.0(LIBVIRT_6.0.0)(64bit)
libvirt.so.0(LIBVIRT_6.10.0)(64bit)
libvirt.so.0(LIBVIRT_7.1.0)(64bit)
libvirt.so.0(LIBVIRT_7.2.0)(64bit)
libvirt.so.0(LIBVIRT_7.3.0)(64bit)
libvirt.so.0(LIBVIRT_7.7.0)(64bit)
libvirt.so.0(LIBVIRT_7.8.0)(64bit)
libvirt.so.0(LIBVIRT_8.0.0)(64bit)
libvirt.so.0(LIBVIRT_PRIVATE_8.1.0)(64bit)
[root@a61c75da8836 /]# ldd usr/local/bin/cloud-api-adaptor 
usr/local/bin/cloud-api-adaptor: /lib64/libvirt-qemu.so.0: version `LIBVIRT_QEMU_8.2.0' not found (required by usr/local/bin/cloud-api-adaptor)
        linux-vdso.so.1 (0x00007ffcafc21000)
        libvirt-lxc.so.0 => /lib64/libvirt-lxc.so.0 (0x00007f4088c2b000)
        libvirt-qemu.so.0 => /lib64/libvirt-qemu.so.0 (0x00007f4088c26000)
        libvirt.so.0 => /lib64/libvirt.so.0 (0x00007f4088600000)
        libc.so.6 => /lib64/libc.so.6 (0x00007f4088200000)
        libselinux.so.1 => /lib64/libselinux.so.1 (0x00007f4088bfa000)
        libglib-2.0.so.0 => /lib64/libglib-2.0.so.0 (0x00007f40884c4000)
        libgcc_s.so.1 => /lib64/libgcc_s.so.1 (0x00007f4088bd8000)
        libgobject-2.0.so.0 => /lib64/libgobject-2.0.so.0 (0x00007f4088b79000)
        libgio-2.0.so.0 => /lib64/libgio-2.0.so.0 (0x00007f4088025000)
        libxml2.so.2 => /lib64/libxml2.so.2 (0x00007f4087eab000)
        libacl.so.1 => /lib64/libacl.so.1 (0x00007f4088b6f000)
        libaudit.so.1 => /lib64/libaudit.so.1 (0x00007f4088b41000)
        libcap-ng.so.0 => /lib64/libcap-ng.so.0 (0x00007f4088b35000)
        libgnutls.so.30 => /lib64/libgnutls.so.30 (0x00007f4087c00000)
        libm.so.6 => /lib64/libm.so.6 (0x00007f4087b22000)
        libnl-3.so.200 => /lib64/libnl-3.so.200 (0x00007f4088b11000)
        libnuma.so.1 => /lib64/libnuma.so.1 (0x00007f4088b03000)
        libyajl.so.2 => /lib64/libyajl.so.2 (0x00007f4088af7000)
        libssh2.so.1 => /lib64/libssh2.so.1 (0x00007f4088480000)
        libssh.so.4 => /lib64/libssh.so.4 (0x00007f4088411000)
        libsasl2.so.3 => /lib64/libsasl2.so.3 (0x00007f4088ad5000)
        libtirpc.so.3 => /lib64/libtirpc.so.3 (0x00007f4087e7b000)
        libcurl.so.4 => /lib64/libcurl.so.4 (0x00007f4087a79000)
        libwsman.so.1 => /lib64/libwsman.so.1 (0x00007f4087e41000)
        libwsman_client.so.5 => /lib64/libwsman_client.so.5 (0x00007f4088403000)
        libwsman_curl_client_transport.so.1 => /lib64/libwsman_curl_client_transport.so.1 (0x00007f4087e38000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f4088c36000)
        libpcre2-8.so.0 => /lib64/libpcre2-8.so.0 (0x00007f40879dc000)
        libpcre.so.1 => /lib64/libpcre.so.1 (0x00007f4087962000)
        libffi.so.8 => /lib64/libffi.so.8 (0x00007f4087956000)
        libgmodule-2.0.so.0 => /lib64/libgmodule-2.0.so.0 (0x00007f4087e31000)
        libz.so.1 => /lib64/libz.so.1 (0x00007f408793c000)
        libmount.so.1 => /lib64/libmount.so.1 (0x00007f40878f5000)
        liblzma.so.5 => /lib64/liblzma.so.5 (0x00007f40878c1000)
        libattr.so.1 => /lib64/libattr.so.1 (0x00007f40878b9000)
        libp11-kit.so.0 => /lib64/libp11-kit.so.0 (0x00007f4087784000)
        libidn2.so.0 => /lib64/libidn2.so.0 (0x00007f4087737000)
        libunistring.so.2 => /lib64/libunistring.so.2 (0x00007f408757f000)
        libtasn1.so.6 => /lib64/libtasn1.so.6 (0x00007f4087567000)
        libnettle.so.8 => /lib64/libnettle.so.8 (0x00007f4087512000)
        libhogweed.so.6 => /lib64/libhogweed.so.6 (0x00007f40874cf000)
        libgmp.so.10 => /lib64/libgmp.so.10 (0x00007f408742a000)
        libssl.so.3 => /lib64/libssl.so.3 (0x00007f4087387000)
        libcrypto.so.3 => /lib64/libcrypto.so.3 (0x00007f4086e00000)
        libgssapi_krb5.so.2 => /lib64/libgssapi_krb5.so.2 (0x00007f4087330000)
        libcrypt.so.2 => /lib64/libcrypt.so.2 (0x00007f40872f6000)
        libkrb5.so.3 => /lib64/libkrb5.so.3 (0x00007f4086d23000)
        libk5crypto.so.3 => /lib64/libk5crypto.so.3 (0x00007f40872dd000)
        libcom_err.so.2 => /lib64/libcom_err.so.2 (0x00007f40872d4000)
        libresolv.so.2 => /lib64/libresolv.so.2 (0x00007f40872c0000)
        libnghttp2.so.14 => /lib64/libnghttp2.so.14 (0x00007f4087295000)
        libpsl.so.5 => /lib64/libpsl.so.5 (0x00007f4087281000)
        libldap.so.2 => /lib64/libldap.so.2 (0x00007f4086cbd000)
        liblber.so.2 => /lib64/liblber.so.2 (0x00007f408726f000)
        libbrotlidec.so.1 => /lib64/libbrotlidec.so.1 (0x00007f4087260000)
        libblkid.so.1 => /lib64/libblkid.so.1 (0x00007f4086c84000)
        libkrb5support.so.0 => /lib64/libkrb5support.so.0 (0x00007f408724f000)
        libkeyutils.so.1 => /lib64/libkeyutils.so.1 (0x00007f4087248000)
        libevent-2.1.so.7 => /lib64/libevent-2.1.so.7 (0x00007f4086c2b000)
        libbrotlicommon.so.1 => /lib64/libbrotlicommon.so.1 (0x00007f4086c08000)
```

It suggests that cloud-api-provider was built on an environment with newer `libvirt-libs` package.

### PR DESCRIPTION
**Pull request by katexochen**
Previously, the docker build used the golang image, which is based
on debian, and copied the dynamically linked binary into a fedora
layer. This change switches the build to use a custom golang-fedora
image as builder.

Fixes #1093
Fixes #1113

Please check if this works for other platforms than amd64.

### CODE REVIEW

**Comment by katexochen** [comment]
@bpradipt @wainersm @stevenhorsman

**Comment by bpradipt** [comment]
Thanks @katexochen . I tried on amd64 without any issues. 

@liudalibj @jtumber-ibm @Amulyam24 can you please try this PR on s390x and ppc64le and confirm it works fine.

**Comment by Amulyam24** [comment]
> @liudalibj @jtumber-ibm @Amulyam24 can you please try this PR on s390x and ppc64le and confirm it works fine.


Hi @katexochen @bpradipt. I successfully verified this on ppc64le. Thanks!

File: Dockerfile
Comment (snir911) [inline Dockerfile:3]: Should we push it to quay.io/kata-containers? maybe even having a CI that pushes it?

**Comment by stevenhorsman** [comment]
I got this working with s390x
My steps were:
- Check out the source branch for the PR: `git checkout -b docker-image-lib katexochen/fix/docker-image-lib`
- Enable buildx for multi-arch: `docker buildx create --use`
- Login to quay.io: `docker login quay.io -u stevenhorsman`
- Run the cloud-api-adaptor release build: ` make ARCHES="linux/amd64,linux/s390x" registry="quay.io/stevenhorsman" RELEASE_BUILD=true image`
- I then used `quay.io/stevenhorsman/cloud-api-adaptor:fae29b1484d190f64df5aa02cbf5481fef411c2e` in my install kustomize and ran the e2e tests that seems to work fine

**Comment by katexochen** [comment]
> Run the cloud-api-adaptor release build:  make ARCHES="linux/amd64,linux/s390x" registry="quay.io/stevenhorsman" RELEASE_BUILD=true image

I guess we need to check with `RELEASE_BUILD=false`, as the binary is otherwise not dynamically linked against libvirt again. Sorry for not mentioning that in the PR description.

**Review by stevenhorsman — COMMENTED** [review]
I think the outstanding question is where we push `golang-fedora:36`. 
Some of the CoCo images have been moved to ghcr, but this was partially done and I think we'd need to get Tobin to add the secrets needed to push to this repo.

We tend to use quay.io/confidential-containers and have the secrets for this set-up in the repo already, so I'd probably vote for that unless we want to migrate everything over to ghcr?

**Comment by katexochen** [comment]
> We tend to use quay.io/confidential-containers and have the secrets for this set-up in the repo already, so I'd probably vote for that unless we want to migrate everything over to ghcr?

I'm a fan of moving everything to GitHub, it will make the future administration easier and the whole project more transparent. But I'm not sure if others think the same and I understand that permissions are a struggle.

We can push it to quay for now and if we want to move to ghcr we can do it in another PR. What should be the name of the image?

**Comment by bpradipt** [comment]
> I'm a fan of moving everything to GitHub, it will make the future administration easier and the whole project more transparent. But I'm not sure if others think the same and I understand that permissions are a struggle.
> 

We can migrate to ghcr in a subsequent PR. Having all artefacts in one place will most likely improve the release and maintenance process. 

> We can push it to quay for now and if we want to move to ghcr we can do it in another PR. What should be the name of the image?

Feel free to suggest something :-) or we can leave it as golang-fedora

**Comment by stevenhorsman** [comment]
> I'm a fan of moving everything to GitHub, it will make the future administration easier and the whole project more transparent. But I'm not sure if others think the same and I understand that permissions are a struggle.

I'm definitely sympathetic to this, but want to have a unifed approach/strategy rather than fragmentation

> We can push it to quay for now and if we want to move to ghcr we can do it in another PR. What should be the name of the image?

`quay.io/confidential-containers/operator:golang-fedora:36` seems fine to me

**Comment by katexochen** [comment]
> `quay.io/confidential-containers/operator:golang-fedora:36` seems fine to me

You meant `quay.io/confidential-containers/cloud-api-adaptor:golang-fedora:36`, right?

**Comment by katexochen** [comment]
I made the registry login and credentials configurable, this way contributors can run the job on their forks if they want. The defaults are the valid input for running the job in this repo and pushing to quay.

**Comment by wainersm** [comment]
Thanks for the fix @katexochen ! Let me test it on my local x86_64 machine before merge, please.

File: hack/Dockerfile.golang
Comment (wainersm) [inline hack/Dockerfile.golang:6]: Any reason to bump go version to 1.20? I am asking because the project is still configured to 1.19.

**Comment by wainersm** [comment]
> > I'm a fan of moving everything to GitHub, it will make the future administration easier and the whole project more transparent. But I'm not sure if others think the same and I understand that permissions are a struggle.
> 
> I'm definitely sympathetic to this, but want to have a unifed approach/strategy rather than fragmentation
> 

1+. I suggest to move to Github registry (when and if) all images together.

> > We can push it to quay for now and if we want to move to ghcr we can do it in another PR. What should be the name of the image?
> 
> `quay.io/confidential-containers/operator:golang-fedora:36` seems fine to me

File: hack/Dockerfile.golang
Comment (katexochen) [reply hack/Dockerfile.golang:6]: There is a difference between the version of the used Go toolchain and the version declared in the go.mod.

The version in the go.mod is a _minimal_ version and specifies which language features are assumed to exist.

Afaik there is no reason ever not to use the latest toolchain, especially as this ensures that security fixes to the standard library are included in the code.

Comment (wainersm) [reply hack/Dockerfile.golang:6]: Sorry, I don't know much about backward compatible in "Gopshere". If there is any problem then picking the latest and greatest stable version (1.120.5) is fine.

File: .github/workflows/build-golang-fedora.yaml
Comment (wainersm) [inline .github/workflows/build-golang-fedora.yaml:7]: What if it additionally gets triggered on push (only and only if the dockerfile changed)? Relying on humans to push the button is always error prone....

**Comment by katexochen** [comment]
I've added the requested changes. Adding a push trigger required some bash, as it is not possible to set inputs. In addition, Fedora an Go version are now parsed from the dockerfile if not otherwise defined, and the Go version is included in the tag.

**Comment by stevenhorsman** [comment]
Nice one, this is looking good to me once the fixups are done :)

**Comment by wainersm** [comment]
I built and push the builder and cloud-api-adaptor images as:

```
$ docker buildx build --platform=linux/amd64 -f hack/Dockerfile.golang -t quay.io/wainersm/cloud-api-adaptor/golang-fedora:36-1.20-amd64 .
$ docker push quay.io/wainersm/cloud-api-adaptor/golang-fedora:36-1.20-amd64
$ sed -i s#quay.io/confidential-containers/cloud-api-adaptor/golang-fedora:1.20.5-36#quay.io/wainersm/cloud-api-adaptor/golang-fedora:36-1.20-amd64# Dockerfile
$ make ARCHES="linux/amd64" registry="quay.io/wainersm/cloud-api-adaptor" RELEASE_BUILD=false image
```
Changed the overlay file:
```
$ cd install/overlays/libvirt/
$ ../../.././kustomize edit set image cloud-api-adaptor=quay.io/wainersm/cloud-api-adaptor:dev-cca53f961f11ecace745b6132aef51910abbb68e-dirty
$ cd -
```

Running as expected:
```
$ kubectl logs pod/cloud-api-adaptor-daemonset-v55gj -n confidential-containers-system
+ exec cloud-api-adaptor libvirt -uri 'qemu+ssh://wmoschet@192.168.122.1/system?no_verify=1' -data-dir /opt/data-dir -pods-dir /run/peerpod/pods -network-name default -pool-name default -socket /run/peerpod/hypervisor.sock
cloud-api-adaptor version v0.6.0-dev
  commit: cca53f961f11ecace745b6132aef51910abbb68e-dirty
  go: go1.20.5
cloud-api-adaptor: starting Cloud API Adaptor daemon for "libvirt"
2023/06/27 22:25:03 [adaptor/cloud/libvirt] libvirt config: &libvirt.Config{URI:"qemu+ssh://wmoschet@192.168.122.1/system?no_verify=1", PoolName:"default", NetworkName:"default", DataDir:"/opt/data-dir"}
2023/06/27 22:25:05 [adaptor] server config: &adaptor.ServerConfig{TLSConfig:(*tlsutil.TLSConfig)(0xc000544000), SocketPath:"/run/peerpod/hypervisor.sock", CriSocketPath:"", PauseImage:"", PodsDir:"/run/peerpod/pods", ForwarderPort:"15150", ProxyTimeout:300000000000}
2023/06/27 22:25:05 [util/k8sops] initialized PeerPodService
2023/06/27 22:25:05 [adaptor] server started
```

**Comment by wainersm** [comment]
@katexochen just curious: will GH autosquash the commits marked with "fixup!"?

**Comment by katexochen** [comment]
> @katexochen just curious: will GH autosquash the commits marked with "fixup!"?

No, it won't. But I really would love them to implement it, I really love fixups. :smile:

**Comment by katexochen** [comment]
> > > > `quay.io/confidential-containers/operator:golang-fedora:36` seems fine to me
> > > 
> > > 
> > > You meant `quay.io/confidential-containers/cloud-api-adaptor:golang-fedora:36`, right?
> > 
> > I suggest to use `golang-fedora:<fedora version>-<golang as Major.Minor version>` (e.g. `golang-fedora:36-1.19`) this way we can update golang version without bump the distro version and don't break older builds. Besides having the golang version on the name makes it more clear...well....the golang version used :)
> 
> Good suggestion ..  Suggesting a minor modification if it's acceptable
> `golang-fedora:<golangversion>-<fedoraversion>` (the tag sequence aligns with the name sequence and easier to decipher )
> 
> 

Had the same thought @bpradipt , and already implemented it like this.

File: .github/workflows/build-golang-fedora.yaml
Comment (wainersm) [inline .github/workflows/build-golang-fedora.yaml:20]: oh, one more thing @katexochen , I promise :)

I created a public `quay.io/wainersm/cloud-api-adaptor`  where I published the built CAA image as `quay.io/wainersm/cloud-api-adaptor/cloud-api-adaptor:dev-cca53f961f11ecace745b6132aef51910abbb68e-dirty`.  When I tried to install CAA on my cluster I got authorization denied to pull the image. I could only get it properly working when I re-published the image in one level down of the repository hierarch (i.e. `quay.io/wainersm/cloud-api-adaptor:dev-cca53f961f11ecace745b6132aef51910abbb68e-dirty`).

That said, I am worried that pushing the builder image to `quay.io/confidential-containers/cloud-api-adaptor/golang-fedora:<golang>-<fedora>` we will hit the same issue.

Comment (katexochen) [reply .github/workflows/build-golang-fedora.yaml]: Updated it to `quay.io/confidential-containers/golang-fedora:<golang>-<fedora>`

Comment (wainersm) [reply .github/workflows/build-golang-fedora.yaml]: > Is this a known limitation of quay.io? I guess another good reason for switching to ghcr...

I don't know. I had never tried to publish to `quay.io/<org>/<repo>/<image>` before.

Comment (stevenhorsman) [reply .github/workflows/build-golang-fedora.yaml]: Yeah - I hit this problem with creating the early coco test images, which I failed to explain properly earlier in this PR

File: Dockerfile
Comment (wainersm) [inline Dockerfile:2]: Becomes `quay.io/confidential-containers/golang-fedora:1.20.5-36`

Comment (wainersm) [reply Dockerfile:2]: I also just created https://quay.io/repository/confidential-containers/golang-fedora?namespace=confidential-containers

Comment (stevenhorsman) [reply Dockerfile]: I gave the robot write access to that repo!

**Review by stevenhorsman — APPROVED** [review]
LGTM. Thanks!

**Comment by wainersm** [comment]
@snir911 last words before I click the "Rebase and merge" button? :)

**Pull request merged**
